src/core/trulens/core/guardrails/base.py

```diff
@@ -98,4 +101,168 @@ def wrapper(*args, **kwargs):
         # note: the following information is manually written to the wrapper because @functools.wraps(func) causes breaking of the method.
         wrapper.__name__ = func.__name__
         wrapper.__doc__ = func.__doc__
+        wrapper.__signature__ = sig
+        return wrapper
+
+
+class block_input:
```

Comment:
class name should be capitalized not snake case

Reply:
Agree in other cases but here this will be the decorator name, matching other existing trulens decorators like `@instrument` and `@context_filter`

Reply:
Note that these are functions in spirit though classes in syntax. They should not be used like classes, hence the naming pattern.

Reply:
makes sense and thanks for the context